gateway-ha/src/main/java/io/trino/gateway/ha/router/HaGatewayManager.java

```diff
@@ -82,13 +83,39 @@ public Optional<ProxyBackendConfiguration> getBackendByName(String name)
     @Override
     public void deactivateBackend(String backendName)
     {
-        dao.deactivate(backendName);
+        updateClusterActivationStatus(backendName, false, () -> dao.deactivate(backendName));
     }
 
     @Override
     public void activateBackend(String backendName)
     {
-        dao.activate(backendName);
+        updateClusterActivationStatus(backendName, true, () -> dao.activate(backendName));
+    }
+
+    private void updateClusterActivationStatus(String clusterName, boolean newStatus, Runnable action)
+    {
+        GatewayBackend model = dao.findFirstByName(clusterName);
+        if (model == null) {
+            throw new IllegalStateException(format("No cluster found with name: %s, could not (de)activate", clusterName));
+        }
+
+        boolean prevStatus = model.active();
+        try {
+            action.run();
+            logActivationStatusChange(clusterName, newStatus, prevStatus);
+        }
+        catch (Exception e) {
+            log.error("Failed to update activation status for cluster %s: %s", clusterName, e.getMessage());
```

Comment:
GatewayResource also logs the error message. I don't think we should log the similar message twice.